packages/metro/src/shared/output/bundle.js

```diff
@@ -26,11 +26,26 @@ function buildBundle(
   map: string,
   ...
 }> {
-  return packagerClient.build({
-    ...Server.DEFAULT_BUNDLE_OPTIONS,
-    ...requestOptions,
-    bundleType: 'bundle',
-  });
+  const OLD_BABEL_ENV = process.env.BABEL_ENV;
+  process.env.BABEL_ENV = requestOptions.dev
+    ? 'development'
+    : process.env.BABEL_ENV || 'production';
+  return packagerClient
+    .build({
+      ...Server.DEFAULT_BUNDLE_OPTIONS,
+      ...requestOptions,
+      bundleType: 'bundle',
+    })
+    .then(
```

Comment:
Could you change this to use `finally` instead? Then you don't need to worry about the return values etc.

(If CI still uses node 8, feel free to bump that to 10+)